app/src/main/java/org/sagebionetworks/research/mpower/room/RoomScheduledActivity.kt

```diff
@@ -0,0 +1,128 @@
+package org.sagebionetworks.research.mpower.room
+
+import android.arch.persistence.room.ColumnInfo
+import android.arch.persistence.room.Embedded
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Index
+import android.arch.persistence.room.PrimaryKey
+import com.google.gson.annotations.SerializedName
+import org.joda.time.DateTime
+import org.sagebionetworks.bridge.rest.model.ActivityType
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+
+import java.util.ArrayList
+
+@Entity
+data class RoomScheduledActivity(@SerializedName("guid") @PrimaryKey var guid: String) {
+    @SerializedName("schedulePlanGuid")
+    var schedulePlanGuid: String? = null
+
+    @SerializedName("startedOn")
+    var startedOn: DateTime? = null
+
+    @SerializedName("finishedOn")
+    @ColumnInfo(index = true)
+    var finishedOn: DateTime? = null
+
+    @SerializedName("scheduledOn")
+    @ColumnInfo(index = true)
+    var scheduledOn: DateTime? = null
+
+    @SerializedName("expiresOn")
+    @ColumnInfo(index = true)
+    var expiresOn: DateTime? = null
+
+    @SerializedName("activity")
+    @Embedded(prefix = "activity_")
+    var activity: RoomActivity? = null
+
+    @SerializedName("persistent")
+    var persistent: Boolean? = null
+
+    @SerializedName("clientData")
+    var clientData: ClientData? = null
+
+    @SerializedName("status")
+    var status: ScheduleStatus? = null
+
+    @SerializedName("type")
+    var type: String? = null
```

Comment:
I believe the type will always be "ScheduledActivity" and is immutable. With that in mind, is it necessary to store this in the RoomDB?